Description: disable optimization of a given function on arm (bug in gcc
 4.4, which creates unittest failures for 64-bit operations)
Author: Iustin Pop <iustin@debian.org>
--- a/src/google/protobuf/wire_format_lite.h
+++ b/src/google/protobuf/wire_format_lite.h
@@ -222,7 +222,11 @@
   static uint32 ZigZagEncode32(int32 n);
   static int32  ZigZagDecode32(uint32 n);
   static uint64 ZigZagEncode64(int64 n);
-  static int64  ZigZagDecode64(uint64 n);
+  static int64  ZigZagDecode64(uint64 n)
+#ifdef __arm__
+    __attribute__((__optimize__(0)))
+#endif
+  ;
 
   // =================================================================
   // Methods for reading/writing individual field.  The implementations
